src/main/java/com/hivemq/bootstrap/HiveMQNettyBootstrap.java

```diff
@@ -95,91 +91,61 @@ private void addDefaultListeners() {
         }
     }
 
-
-    @NotNull
-    private List<BindInformation> bindTcpListeners(final @NotNull List<TcpListener> tcpListeners) {
+    private @NotNull List<BindInformation> bindTcpListeners(final @NotNull List<TcpListener> tcpListeners) {
         log.trace("Checking TCP listeners");
         final ImmutableList.Builder<BindInformation> futures = ImmutableList.builder();
-        for (final TcpListener tcpListener : tcpListeners) {
-
-            final ServerBootstrap b = createServerBootstrap(nettyConfiguration.getParentEventLoopGroup(), nettyConfiguration.getChildEventLoopGroup(), tcpListener);
-            final String bindAddress = tcpListener.getBindAddress();
 
-            final Integer port = tcpListener.getPort();
+        for (final TcpListener listener : tcpListeners) {
 
-
-            log.info("Starting TCP listener on address {} and port {}", bindAddress, port);
-            final ChannelFuture bind = b.bind(createInetSocketAddress(bindAddress, port));
-            futures.add(new BindInformation(tcpListener, bind));
+            final ServerBootstrap b = createServerBootstrap(nettyConfiguration.getParentEventLoopGroup(), nettyConfiguration.getChildEventLoopGroup(), listener);
+            log.info("Starting TCP listener on address {} and port {}", listener.getBindAddress(), listener.getPort());
+            final ChannelFuture bind = b.bind(listener.getBindAddress(), listener.getPort());
```

Comment:
does this the same as b.bind(createInetSocketAddress(bindAddress, port)); ?

Reply:
below it was already done this way, just a double check

Reply:
Netty does this already internally. I streamlined the code to be the same with the others.